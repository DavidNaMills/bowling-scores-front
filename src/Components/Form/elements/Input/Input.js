import React from 'react';
import classes from './Input.module.scss';
import { CreateSharp, ErrorSharp } from '@material-ui/icons'

const iconOk = [classes.icon, classes.icon__required];
const iconErr = [classes.icon, classes.icon__error];

const Input = ({ changed, id, type = 'text', label = null, placeholder = '', error = [], isValid = false, isRequired = false }) => {
    let styles = [classes.input, classes.input__default]

    if (isValid) {
        styles = styles.concat(classes.input__valid);
    }

    if (error.length > 0) {
        styles = styles.concat(classes.input__error);
    }

    return (
        <div className={classes.input__label}>
            {label && <label htmlFor={id}>{label}</label>}
            <div>
                <input
                    className={styles.join(' ')}
                    id={id}
                    type={type}
                    onChange={e => changed(e, id)}
                    placeholder={placeholder}
                />

                {
                    error.length > 0
                        ? <ErrorSharp className={iconErr.join(' ')} />
                        : isRequired && <CreateSharp className={iconOk.join(' ')} />
                }
            </div>
            {error.length > 0 &&
                <div className={classes.input__errorContainer}>
                    {
                        error.map((x, i) => <p key={i} className={classes.input__errorContainer_errorMsg}>{x}</p>, 0)
                    }
                </div>
            }
        </div>
    )
}

export default Input;