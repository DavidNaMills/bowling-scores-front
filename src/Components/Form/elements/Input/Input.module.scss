@import '../../../../styles/colors.scss';
@import '../../../../styles/fonts.scss';

.input{
    border: none;
    padding: 5px;
    width: $input_width;
    
    &__label{
        margin-top: $input_spacing;
    }

    &__default{
        background: rgb(230, 230, 230);
    }

    &__valid{
        -webkit-box-shadow: $valid_input_shadow;
        -moz-box-shadow: $valid_input_shadow;
        box-shadow: $valid_input_shadow;
        border-bottom: solid 1px $btn_default_bkg;
    };
    
    &__error{
        -webkit-box-shadow: $error_input_shadow;
        -moz-box-shadow: $error_input_shadow;
        box-shadow: $error_input_shadow;
        border-bottom: solid 1px $btn_danger_bkg;
    }

    &__errorContainer{
        width: $input_width;
        padding: 5px;

        -webkit-box-shadow: $error_input_shadow;
        -moz-box-shadow: $error_input_shadow;
        box-shadow: $error_input_shadow;

        &_errorMsg{
            font-size: $error_size_small;
            color: $btn_danger_bkg;
        }
    }
}

.icon{
    margin-left: 5px;
    display: flex;
    justify-content: center;
    align-content: center;

    &__required{
        color: $btn_default_bkg_active_hover;
    }

    &__error{
        color: $btn_danger_bkg_active;
    }
}