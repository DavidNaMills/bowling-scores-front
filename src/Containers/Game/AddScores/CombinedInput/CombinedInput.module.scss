.combinedInput{
    display: flex;
    justify-content: space-around;
    align-items: center;
    // padding: 10px;

    &__name{
        width: 60%;
        padding: 0;
        margin: 0;
    }
    
    &__input{
        justify-content: left;
        width: 40%;
    }
}