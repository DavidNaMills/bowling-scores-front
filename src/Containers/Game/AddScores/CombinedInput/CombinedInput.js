import React from 'react';
import classes from './CombinedInput.module.scss';

import Input from '../../../../Components/Form/elements/Input/Input';
import { playerTableStyle } from '../../playerTableStyle';

//Avatar when built

const CombinedInput = ({ name, color, value, updateScore, id }) => {

    // <div style={playerTableStyle()}>
    return (
        <div style={playerTableStyle(color)} className={classes.combinedInput}>
            <div className={[classes.combinedInput__name, classes.combinedInput].join(' ')}>
                <p>{name}</p>
            </div>

            <div className={[classes.combinedInput__input, classes.combinedInput].join(' ')}>
                <Input
                    value={value}
                    changed={e => updateScore({ id, data: e.target.value })}
                    id={id}
                    type='number'
                    max={300}
                    min={0}
                />
            </div>
        </div>
    )
}

export default CombinedInput;