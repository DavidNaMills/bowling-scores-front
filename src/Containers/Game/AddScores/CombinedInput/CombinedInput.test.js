import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import CombinedInput from './CombinedInput';
import Input from '../../../../Components/Form/elements/Input/Input';


const updateScoreMock = jest.fn();
const testConfig = {
    name: 'David',
    color: '123, 123, 123',
    value: 123,
    updateScore: { updateScoreMock },
    id: 'f2d3fd3ds'
}


describe('<CombinedInput /> test suite', () => {
    describe('snapshot tests', () => {
        test('matches default snapshot', () => {
            const component = renderer.create(<CombinedInput {...testConfig}/>);
            const app = component.toJSON();
            expect(app).toMatchSnapshot();
        });
    });

    describe('render tests', ()=>{
        const wrapper = mount(<CombinedInput {...testConfig}/>)

        it('displays the players name', ()=>{
            expect(wrapper.find('p').text()).toEqual(testConfig.name);
            expect(wrapper.find('.combinedInput').length).toBe(3);
            expect(wrapper.find('.combinedInput__name').length).toBe(1);
        });

        it('contains an <Input/> component with div wrappers', ()=>{
            expect(wrapper.find('.combinedInput__input').length).toBe(1);
            expect(wrapper.find(Input).length).toBe(1);
        });
    });
});