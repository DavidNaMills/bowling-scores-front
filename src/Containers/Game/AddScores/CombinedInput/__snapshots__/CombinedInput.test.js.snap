// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CombinedInput /> test suite snapshot tests matches default snapshot 1`] = `
<div
  className="combinedInput"
  style={
    Object {
      "background": "linear-gradient(180deg, rgba(255,255,255,1) 37%, rgba(123, 123, 123,1) 100%)",
    }
  }
>
  <div
    className="combinedInput__name combinedInput"
  >
    <p>
      David
    </p>
  </div>
  <div
    className="combinedInput__input combinedInput"
  >
    <div
      className="input__label"
    >
      <div>
        <input
          className="input input__default"
          id="f2d3fd3ds"
          max={300}
          min={0}
          onChange={[Function]}
          placeholder=""
          type="number"
          value={123}
        />
      </div>
    </div>
  </div>
</div>
`;
